docs/docs/specs/v0.2/session-graph.md

```diff
@@ -0,0 +1,292 @@
+  }
+  
+  // Creates the one-to-one parent relations from the content nodes back to the group node
+  const provenanceParentRelations = documents.reduce((acc, document)=>{
+    acc.push({of: document.id, by: MARKDOWN_GROUP, key: 'provenance-parent'})
+  }, []);
+  
+  // Adds both kinds of relation to the graph
+  sessionGraph.addRelations(provenanceChildrenRelation, ...provenanceParentRelations);
+  
+  // Done!
+}
+```
+
+#### Rendering resolved data & child nodes
+
+```tsx
+const ListItem = ({ id, node }: {id: NodeKey, node: Node}) => {
+  const datum = useResolvedDatum(id);
+  return <Surface role='listitem' datum={datum} />
+}
+
+const List = ({id, node, relationKey = 'provenance-child'}: {id, NodeKey, node: Node, relationKey: string}) => {
+  const datum = useResolvedDatum(id);
+  const childNodes: Map<NodeKey, Node> = useRelatedNodes(id, relationKey);
+  return <>
+    <Surface role='list__meta' datum={datum} />
+    <ul>
+      {childNodes.entries().map((childId, node)=>(
+        <ListItem id={childId} node={node}/>
+      ))}
+    </ul>
+  </>
+}
+```
+
+The hooks `useResolvedDatum` and `useRelatedNodes` leverage signals to propagate updates to the substrate entity or the node’s relations respectively.
+
+React easily memoizes `node.id` and `relationValue` strings, so, when these change, the component tree is reevaluated at the right scope.
+
+### Preferential relations
+
+Plugins should be able to add a node that relates to other plugins’ content items in a way that can take precedence in some situations.
+
+e.g.
+> `GithubPlugin` can “claim” under its node the nodes added by `MarkdownPlugin` that have certain parameters `GithubPlugin` recognizes.
+
+#### Adding preferential relations
+
+```tsx
+const GH_NS = 'dxos.org/github-plugin';
+
+const GH_GROUP = `${MARKDOWN_PLUGIN}/documents`;
+
+const handleReady = async () => {
+    
+  // The plugin creates its preferential group node
+  const groupNode: Node = {
+    id: GH_GROUP,
+    label: ['plugin name', {ns: GH_NS}],
+    params: {
+      from: GH_NS,
+      icon: 'GithubLogo'
+    },
+  };
+    
+    // Gets the entities from its ontological substrate
+  const ghDocuments = await runQuery(GithubMarkdownSessionGraphQuery);
+  
+  const navmenuChildrenRelation = {
+    of: MARKDOWN_GROUP,
+    by: new Set(ghDocuments.map(({id})=>id)),
+    key: 'navmenu-child'
+  }
+  
+    // Creates the one-to-one parent relations from the content nodes back to the group node
+  const navmenuParentRelations = documents.reduce((acc, document)=>{
+    acc.push({of: document.id, by: MARKDOWN_GROUP, key: 'navmenu-parent'})
+  }, []);
+  
+}
+```
+
+#### Rendering lists by preferential relation
+
+```tsx
+
+const SeparatedLists = ({id, root}: {id: NodeKey, root: Node}) => {
+  const datum = useResolvedDatum(id);
+  const specialChildren = useRelatedNodes(id, 'navmenu-child');
+  const specialIds = new Set(specialChildren.keys());
+  const allChildren = useRelatedNodes(id, 'provenance-child');
```

Comment:
This is a little confusing to me, perhaps just slightly too abstract, I can't quite picture what each of these would contain to facilitate rendering the tree as expected.

Reply:
You mean the implementation of the hooks would contain, or what these hooks are returning?

If the former, I want to illustrate that. If the latter, they’re returning `Map<NodeKey, Node>`

Reply:
I mean more just a concrete example, like an example set of nodes and what would be returns if their ids were passed in.

Reply:
`useRelatedNodes` takes as arguments the one _subject Node’s_ `NodeKey`, then a `RelationKey`. This use-case gets ids from `specialChildren.keys()` so it can do set math.

Is there something specific you have in mind you’re wondering about?